lib/util.cpp

```diff
@@ -93,6 +97,26 @@ QString QMatrixClient::cacheLocation(const QString& dirName)
     return cachePath;
 }
 
+qreal QMatrixClient::stringToHueF(const QString &string)
+{
+    Q_ASSERT(!string.isEmpty());
+    QByteArray hash = QCryptographicHash::hash(string.toUtf8(),
+                                               QCryptographicHash::Sha1);
+    QDataStream dataStream(qToLittleEndian(hash).left(2));
+    dataStream.setByteOrder(QDataStream::LittleEndian);
+    quint16 hashValue;
+    dataStream >> hashValue;
+    const auto hueF =
+            qreal(hashValue)/std::numeric_limits<quint16>::max();
+    Q_ASSERT((0 <= hueF) && (hueF <= 1));
+    return hueF;
+}
+
+int QMatrixClient::hueFtoHue(qreal hueF)
+{
+    return int(hueF*359);
+}
```

Comment:
I'd rather not generalise too aggressively - for different cases there can be different preferred discretisation ratios. Let's keep this within `User::hue()` for now (even that will need to be field-proven at least until the next library release).
Alternatively, if you really want to provide an API aligned with that of Qt, it shouldn't be `hueFtoHue()` but rather `stringToHue()` that internally calls `stringToHueF()`. But for now I wouldn't even bother; a `qreal` version is sufficient for all interactions with Qt.